src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,67 @@
+package duke.parser;
+
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.enums.ErrorCodes;
+import duke.enums.TaskTypes;
+import duke.exceptions.DukeException;
+
+public class Parser {
+    /**
+     * Parses the full command input from the user and returns the appropriate command to be execute by caller.
+     *
+     * @param fullCommand String of the full command input by the user.
+     * @return A Command object based on the type of command parsed.
+     * @throws DukeException DukeException thrown if missing task information or unknown commands.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        Command command = null;
+        switch (fullCommand) {
+        case "list":
+            command = new ListCommand();
+            break;
+        case "bye":
+            command = new ExitCommand();
+            break;
+        default:
+            String[] splitString = fullCommand.split(" ", 2);
+            if (splitString.length != 2) {
+                throw new DukeException(ErrorCodes.MISSING_TASK_NAME);
+            }
+            String userCommand = splitString[0];
+            String userArgs = splitString[1];
```

Comment:
What if a new command which requires some other number of arguments is required?
Consider splitting the nested `switch` statements into multiple sequential `switch` statements.